atlasexec/atlas_test.go

```diff
@@ -420,6 +420,130 @@ func TestMigrateLintWithLogin(t *testing.T) {
+					Dir    string `json:"dir"`
+				} `json:"input"`
+			}
+			SyncDir *struct {
+				Input struct {
+					Slug    string                `json:"slug"`
+					Driver  string                `json:"driver"`
+					Dir     string                `json:"dir"`
+					Context *atlasexec.RunContext `json:"context"`
+				} `json:"input"`
+			}
+		}
+		httpTest struct {
+			payloads []graphQLQuery
+			srv      *httptest.Server
+		}
+	)
+	token := "123456789"
+	newHttpTest := func() (*httpTest, string) {
+		tt := &httpTest{}
+		handler := func() http.HandlerFunc {
+			return func(w http.ResponseWriter, r *http.Request) {
+				require.Equal(t, "Bearer "+token, r.Header.Get("Authorization"))
+				var query graphQLQuery
+				require.NoError(t, json.NewDecoder(r.Body).Decode(&query))
+				tt.payloads = append(tt.payloads, query)
+				if strings.Contains(query.Query, "pushDir") {
+					fmt.Fprint(w, `{"data":{"pushDir":{"url":"https://some-org.atlasgo.cloud/dirs/314159/tags/12345"}}}`)
+				}
+				if strings.Contains(query.Query, "syncDir") {
+					fmt.Fprint(w, `{"data":{"syncDir":{"url":"https://some-org.atlasgo.cloud/dirs/314159/tags/12345"}}}`)
+				}
+			}
+		}
+		tt.srv = httptest.NewServer(handler())
+		t.Cleanup(tt.srv.Close)
+		return tt, generateHCL(t, token, tt.srv)
+	}
+	c, err := atlasexec.NewClient(".", "atlas")
+	require.NoError(t, err)
+	inputContext := &atlasexec.RunContext{
+		Repo:   "testing-repo",
+		Path:   "path/to/dir",
+		Branch: "testing-branch",
+		Commit: "sha123",
+		URL:    "this://is/a/url",
+	}
+	t.Run("sync", func(t *testing.T) {
+		params := &atlasexec.MigratePushParams{
+			DevURL: "sqlite://file?mode=memory",
+			DirURL: "file://testdata/migrations",
+			Name:   "test-dir-slug",
+			Env:    "test",
+		}
+		t.Run("with context", func(t *testing.T) {
+			tt, atlasConfigURL := newHttpTest()
+			params.ConfigURL = atlasConfigURL
+			got, err := c.MigratePush(context.Background(), params)
+			require.NoError(t, err)
+			require.NotEqual(t, "1234", got)
+			require.Len(t, tt.payloads, 2)
+			p := &tt.payloads[1]
+			require.Contains(t, p.Query, "syncDir")
+			err = json.Unmarshal(p.Variables, &p.SyncDir)
+			require.NoError(t, err)
+			require.Equal(t, "test-dir-slug", p.SyncDir.Input.Slug)
+			require.Equal(t, "SQLITE", p.SyncDir.Input.Driver)
+			require.NotEmpty(t, p.SyncDir.Input.Dir)
+		})
+		t.Run("without context", func(t *testing.T) {
+			tt, atlasConfigURL := newHttpTest()
+			params.ConfigURL = atlasConfigURL
+			params.Context = inputContext
+			got, err := c.MigratePush(context.Background(), params)
+			require.NoError(t, err)
+			require.NotEqual(t, "1234", got)
+			require.Len(t, tt.payloads, 2)
+			p := &tt.payloads[1]
+			require.Contains(t, p.Query, "syncDir")
+			err = json.Unmarshal(p.Variables, &p.SyncDir)
+			require.NoError(t, err)
+			require.Equal(t, "test-dir-slug", p.SyncDir.Input.Slug)
+			require.Equal(t, "SQLITE", p.SyncDir.Input.Driver)
+			require.NotEmpty(t, p.SyncDir.Input.Dir)
```

Comment:
My suggestion - no need to copy-paste everything. Test only the `context` change. 
```suggestion
```